### PR TITLE
underwater detection based on sky object

### PR DESCRIPTION
**Pull request by mchaptel**
use the sky TES object flags to detect whether to activate underwater

### CODE REVIEW

File: TESReloaded/Framework/NewVegas/Game.h [hunk 1]
```diff
@@ -4060,6 +4060,23 @@ class Sky {
 	float GetSunriseColorBegin() { return ThisCallD(0x63B9B0, this); }
 	float GetSunsetColorEnd() { return ThisCallD(0x63BA30, this); }
 
+	enum Flags : unsigned __int32
```
Comment (llde) [inline TESReloaded/Framework/NewVegas/Game.h:4063]: Use UInt32

**Review by llde — CHANGES_REQUESTED** [review]
Replicate the IsUnderwater to Oblivion frameework too,

File: TESReloaded/Framework/NewVegas/Game.h (hunk 1 above)
Comment (llde) [reply TESReloaded/Framework/NewVegas/Game.h:4063]: > can you check that it is in fact supported? I don't have oblivion to test it. I'll make the changes and you can test them if you prefer

**Comment by llde** [comment]
change unsigned __int32 to UInt32 @mchaptel and I commit

**Comment by mchaptel** [comment]
I've made the change and also added the relevant flags for oblivion. I figured it would be better to fix it if the flag is wrong instead of having it break because the function is called without existing?

I've also commented out the code that makes the reflection fix happen only when the water shader is active. Let me know if I need to remove it

**Review by llde — APPROVED** [review]

**Pull request merged**